docs/dgeni.conf.js

```diff
@@ -31,12 +33,26 @@ package.config(function(log, readFilesProcessor, templateFinder, writeFilesProce
     basePath: 'src'
   }];
 
+
+  // Nunjucks and Angular conflict in their template bindings so change the Nunjucks
+  templateEngine.config.tags = {
+    variableStart: '{$',
+    variableEnd: '$}'
+  };
+
+  templateEngine.filters.push(require('./rendering/relativeLinkInlineTag'));
+
   // Add a folder to search for our own templates to use when rendering docs
   templateFinder.templateFolders.unshift(path.resolve(__dirname, 'templates'));
 
   // Specify how to match docs to templates.
   // In this case we just use the same static template for all docs
-  templateFinder.templatePatterns.unshift('common.template.html');
+  templateFinder.templatePatterns = [
+    '${ doc.template }',
+    '${ doc.id }.${ doc.docType }.template.html',
+    '${ doc.id }.template.html',
+    '${ doc.docType }.template.html',
+    'common.template.html'];
```

Comment:
This is not entirely needed but these are a good set of fallback patterns for matching docs to their templates.
In the pull request we are using `${ doc.docType }.template.html` to match the index doc to its template; we are then falling back to `common.template.html` for all the other docs (i.e. all the js docTypes).